### ISSUE TITLE
An OSError about header when importing neuralynx data

### ISSUE THREAD

**Issue by redagavin** [NeuralEnsemble/python-neo#1202]
I was trying to import neuralynx data into python using neo. But, I got an OSError.
This is a screenshot of the error.
![Screen Shot 2022-11-27 at 5 07 43 PM](https://user-images.githubusercontent.com/39891180/204162273-7ef68c45-302c-4e80-bfd6-bac6da9e915b.png)

Also, I noticed that there is a small bug in the line that raise the OSError. I think `.format(an, av)` is supposed to fill the first {} with `an` and the second {} with `av`. However, only the second string after the `+` calls `.format(an, av)` in the source code, which is why the error message is weird.
<img width="800" alt="Screen Shot 2022-11-27 at 5 19 11 PM" src="https://user-images.githubusercontent.com/39891180/204162647-a07972fd-f759-4677-8209-702cb060a4aa.png">

**Comment by redagavin** [comment]
@samuelgarcia @JuliaSprenger @apdavison Could you please take a look at this? Thanks!

**Comment by samuelgarcia** [comment]
Hi,
thanks for the report.
Could you send us a file to debug this ?

@JuliaSprenger @PeterNSteinmetz: would you have time to handle this, you are our best neuralynx expert!

**Comment by redagavin** [comment]
> Hi, thanks for the report. Could you send us a file to debug this ?
> 
> @JuliaSprenger @PeterNSteinmetz: would you have time to handle this, you are our best neuralynx expert!

@samuelgarcia @JuliaSprenger @PeterNSteinmetz Yes, I can send you a file to debug this. Please provide me with a link so that I can share the file privately.

**Comment by PeterNSteinmetz** [comment]
Gavin, I can take a look at that. Can you send via email to PeterNSteinmetz@steinmetz.org ? Alternatively, do you use Google Drive? If so you could share with me there - username is PeterNSteinmetz

**Comment by redagavin** [comment]
> Gavin, I can take a look at that. Can you send via email to [PeterNSteinmetz@steinmetz.org](mailto:PeterNSteinmetz@steinmetz.org) ? Alternatively, do you use Google Drive? If so you could share with me there - username is PeterNSteinmetz

I sent the files to you through email and google drive. Please take a look. Thanks!

**Comment by samuelgarcia** [comment]
Thank you Peter!

**Comment by PeterNSteinmetz** [comment]
I now have a fix for this issue. New undocumented header strings for Cheetah v 5.6.0 caused this. I want to add the test data to ephy_testing_data. 

So @redagavin, can you please provide permission and desired credit/attribution for including  one shortened .ncs file (26.8 kB) and one shortened .ntt file (22.4 kB) from what you sent me to the repository at https://gin.g-node.org/NeuralEnsemble/ephy_testing_data

**Comment by PeterNSteinmetz** [comment]
@samuelgarcia @JuliaSprenger Also will need some help here again placing the data in the ephy_testing_data repository. (Sorry but this whole gin thing is not very clear to me because I usually think in git terms). 

I have my own clone of this project and have that checked out and synced and have a master branch on the web site. What do I need to do to make sure that my master is up to date with the project master? I don't see any sort of sync button. 

Then I need to create a branch, unlock the Neuralynx folder, add a new sub-folder with two files, lock it and then commit and push to my branch on my repository. Then open a pull request on the project master? 

Is there somewhere where this sort of adding things to an already existing clone is laid out?

**Comment by PeterNSteinmetz** [comment]
@samuelgarcia @JuliaSprenger We have no word yet on the use of the sample files for testing. How do you want to proceed? 

I could just create a code branch for Neo with the fixes but without any tests. 

Alternately, I could hack the headers on these a bit to anonymize them and either remove all the data records or replace the data with random samples. The test files would then be completely synthetic, but not as realistic.

**Comment by JuliaSprenger** [comment]
> I have my own clone of this project and have that checked out and synced and have a master branch on the web site. What do I need to do to make sure that my master is up to date with the project master? I don't see any sort of sync button.

If you are using the gin-cli simply `gin download` will update your local copy of the branch you have currently checked out.
> 
> Then I need to create a branch, unlock the Neuralynx folder, add a new sub-folder with two files, lock it and then commit and push to my branch on my repository. Then open a pull request on the project master?
For adding new files you don't need to unlock existing files (folders can not be in a locked state, only files can). Also note that since git annexed files are not shared between forks, you should push your feature branch directly to the `Neuralensemble/ephy_testing_data` repo and not your fork. You should have permission to do so as you are already listed as a collaborator.
To be sure the gin-cli runs smoothly ensure that you are using a git-annex<9 and not the latest version, since we noticed some issues with that version (see also [this issue](https://github.com/G-Node/gin-cli/issues/319))

Tell me if you have further question following the [contribution guideline](https://gin.g-node.org/NeuralEnsemble/ephy_testing_data/src/master/README.md)

**Comment by PeterNSteinmetz** [comment]
@JuliaSprenger Actually there is the issue noted here in the Jan 23 comment. We don't have permission to use the test files which were given.

**Comment by JuliaSprenger** [comment]
@redagavin Could you comment on this?

> So @redagavin, can you please provide permission and desired credit/attribution for including one shortened .ncs file (26.8 kB) and one shortened .ntt file (22.4 kB) from what you sent me to the repository at https://gin.g-node.org/NeuralEnsemble/ephy_testing_data

**Comment by PeterNSteinmetz** [comment]
Well see if that produces anything. I have been in touch with him via email to get the files and wrote and received nothing.

**Comment by GeezleCode** [comment]
Hi, I solved a similar bug in my fork. The issue was that neo didn't recognize the "RawDataFile" option in the headerstring. This is set when ".ncs" files are created by offline replay of raw data ".nrd" files. You'll find the fork here https://github.com/GeezleCode/python-neo. The branch containing the few changes is called "NeuralynxIORawDataSystem". I haven't created a pull request yet.

**Comment by JuliaSprenger** [comment]
Hi @GeezleCode Thanks for fixing another issue of the NeuralynxIO. If you want your fix to be included in neo, could you open a PR?

**Comment by GeezleCode** [comment]
@JuliaSprenger Done! #1234

**Comment by JuliaSprenger** [comment]
@PeterNSteinmetz If it's just a small change we could add it anyway. Without test files we can just not guarantee that it will not reoccur at some point in the future...

**Comment by PeterNSteinmetz** [comment]
Pull request 1257 should handle this case now.
